### PR TITLE
feat: support function for `BannerPlugin`

### PR DESCRIPTION
**Pull request by ahabhgk**
## Summary

- support function for `BannerPlugin`
- replace `lib/BannerPlugin.js` with builtin plugin BannerPlugin
- delete unnecessary code in `lib`, hope we can totally remove `lib` someday

## Test Plan

- pass webpack-test banner-plugin

## Require Documentation?

- [ ] No
- [x] Yes, ~~the corresponding rspack-website PR is \_\_~~ I will add docs after docs of builtin plugins re-merged

### CODE REVIEW

File: crates/node_binding/src/js_values/mod.rs [hunk 1]
```diff
@@ -1,5 +1,9 @@
+mod chunk {
+  // TODO: should we merge rspack_binding_options and node_binding?
```
Comment (ahabhgk) [inline crates/node_binding/src/js_values/mod.rs:2]: cc @h-a-n-a

Comment (h-a-n-a) [reply crates/node_binding/src/js_values/mod.rs:2]: > or `rspack_plugin_xxx` export `RawXxxOptions` under feature `node_binding`

This looks good to me, keeping these options in the same crate.

Comment (h-a-n-a) [reply crates/node_binding/src/js_values/mod.rs:2]: Just for heads-up. @Boshen seems treat every undefined symbol to lookup at runtime. For convenience, you can omit the feature. But still, I would love to add one though.

Comment (ahabhgk) [reply crates/node_binding/src/js_values/mod.rs:2]: > or `rspack_plugin_xxx` export `RawXxxOptions` under feature `node_binding`

I'm not familiar with napi or cargo or some build stuff, not sure if it's possible or good for ci build time, so you guys can decide it

Comment (h-a-n-a) [reply crates/node_binding/src/js_values/mod.rs:2]: Let's stick to feature, it's a common practice and do not coupled with linker options.

File: packages/rspack/src/builtin-plugin/ProgressPlugin.ts [hunk 2]
```diff
@@ -1,8 +1,8 @@
 import { RawProgressPluginConfig } from "@rspack/binding";
 import { BuiltinPluginKind, create } from "./base";
 
-export type ProgressPluginOptions = RawProgressPluginConfig | undefined;
+export type ProgressPluginArgument = RawProgressPluginConfig | undefined;
```
Comment (h-a-n-a) [inline packages/rspack/src/builtin-plugin/ProgressPlugin.ts:4]: Do we have to even align with the type name? For consistency I would love to use `XXXOptions`

**Review by h-a-n-a — APPROVED** [review]

File: packages/rspack/src/builtin-plugin/ProgressPlugin.ts (hunk 2 above)
Comment (ahabhgk) [reply packages/rspack/src/builtin-plugin/ProgressPlugin.ts:4]: I also prefer `XXXOptions`, but types in webpack are not consistent..., let's keep the type alignment for now, at least for facade type

**Pull request merged**